autogpt_platform/backend/backend/integrations/webhooks/graph_lifecycle_hooks.py

```diff
@@ -96,105 +100,25 @@ async def on_node_activate(
+            user_id=user_id,
+            trigger_block=node.block,
+            trigger_config=node.input_default,
+            for_graph_id=node.graph_id,
         )
-
-    logger.debug(
-        f"Activating webhook node #{node.id} with config {block.webhook_config}"
-    )
-
-    webhooks_manager = get_webhook_manager(provider)
-
-    if auto_setup_webhook := isinstance(block.webhook_config, BlockWebhookConfig):
-        try:
-            resource = block.webhook_config.resource_format.format(**node.input_default)
-        except KeyError:
-            resource = None
-        logger.debug(
-            f"Constructed resource string {resource} from input {node.input_default}"
-        )
-    else:
-        resource = ""  # not relevant for manual webhooks
-
-    block_input_schema = cast(BlockSchema, block.input_schema)
-    credentials_field_name = next(iter(block_input_schema.get_credentials_fields()), "")
-    credentials_meta = (
-        node.input_default.get(credentials_field_name)
-        if credentials_field_name
-        else None
-    )
-    event_filter_input_name = block.webhook_config.event_filter_input
-    has_everything_for_webhook = (
-        resource is not None
-        and (credentials_meta or not credentials_field_name)
-        and (
-            not event_filter_input_name
-            or (
-                event_filter_input_name in node.input_default
-                and any(
-                    is_on
-                    for is_on in node.input_default[event_filter_input_name].values()
-                )
-            )
-        )
-    )
-
-    if has_everything_for_webhook and resource is not None:
-        logger.debug(f"Node #{node} has everything for a webhook!")
-        if credentials_meta and not credentials:
-            raise ValueError(
-                f"Cannot set up webhook for node #{node.id}: "
-                f"credentials #{credentials_meta['id']} not available"
-            )
-
-        if event_filter_input_name:
-            # Shape of the event filter is enforced in Block.__init__
-            event_filter = cast(dict, node.input_default[event_filter_input_name])
-            events = [
-                block.webhook_config.event_format.format(event=event)
-                for event, enabled in event_filter.items()
-                if enabled is True
-            ]
-            logger.debug(f"Webhook events to subscribe to: {', '.join(events)}")
+        if new_webhook:
+            node = await set_node_webhook(node.id, new_webhook.id)
         else:
-            events = []
-
-        # Find/make and attach a suitable webhook to the node
-        if auto_setup_webhook:
-            assert credentials is not None
-            new_webhook = await webhooks_manager.get_suitable_auto_webhook(
-                user_id,
-                credentials,
-                block.webhook_config.webhook_type,
-                resource,
-                events,
+            logger.debug(
```

Comment:
We need a way to surface this issue to the user long term probs

Reply:
If there’s a way this error can cause a user’s graph to not run when they expect it should be told to them imo

it’s not just this specific case that it’s a problem to be addressed but holistically